Tests/SwiftDocCTests/Infrastructure/ReferenceResolverTests.swift

```diff
@@ -735,6 +736,117 @@ class ReferenceResolverTests: XCTestCase {
             assertion(resolvedSymbol)
         }
     }
+    
+    func testEmitsDiagnosticsForEachDocumentationChunk() throws {
+        let moduleReference = ResolvedTopicReference(bundleIdentifier: "com.example.test", path: "/documentation/ModuleName", sourceLanguage: .swift)
+        let reference = ResolvedTopicReference(bundleIdentifier: "com.example.test", path: "/documentation/ModuleName/Something", sourceLanguage: .swift)
+        
+        let inSourceComment = """
+        Some description of this class
+        
+        These links to ``NotFoundSymbol`` and <doc:NotFoundArticle> won't resolve.
+        
+        This image name won't resolve: ![Some image that's not found](not-found-image)
+        """
+        let start = (line: 7, character: 4) // arbitrary non-zero values
+        let sourceCodeURL = URL(fileURLWithPath: "/Users/username/path/to/Something.swift")
+        
+        let symbol = SymbolGraph.Symbol(
```

Comment:
This code to create a symbol from a certain doc comment looks super-useful. Might be nice to extract this into a generic helper function any test could use.

Reply:
We have lots of test helpers that create symbols. I'd be good to consolidate them but I think that's for another PR. 

Even this doc-comment creation code exist in 3 places now (which by my rule-of-thumb is when it's time to think about a common API).

Reply:
I opened a #955 which combines many of our various SGF test helpers.